packages/hoppscotch-common/src/helpers/actions.ts

```diff
@@ -65,6 +65,9 @@ type HoppActionArgsMap = {
   "rest.request.open": {
     doc: HoppRESTDocument
   }
+  "request.save-as": {
+    request: HoppRESTRequest | HoppGQLRequest
+  }
```

Comment:
I would suggest the unions be discriminated based on a `requestType` field to make it easier to validate. Either that or you should split this event for `rest.request.save-as` and `gql.request.save-as`.

```suggestion
  "request.save-as":
  	| {
    		requestType: "rest"
    		request: HoppRESTRequest
  	  }
  	| {
  		requestType: "gql"
  		request: HoppGQLRequest
  	  } 
```